packages/dynamite/dynamite/lib/src/models/type_result/type_result.dart

```diff
@@ -107,13 +109,13 @@ sealed class TypeResult {
       ..write(serializerName ?? 'jsonSerializers')
       ..write('.deserialize(')
       ..write(object)
-      ..write(', specifiedType: const $fullType)');
+      ..write(', specifiedType: $fullType)');
 
     if (!nullable) {
       buffer.write('!');
     }
 
-    return '($buffer as $name)';
+    return '$buffer as $name';
```

Comment:
This is usually needed for safety. If someone would call `.toString()` or anything else on it it will result in broken code. I assume this never happens as otherwise the code in this PR would be broken?

Reply:
it only happens when validating a serialized someOf (we did need it before as the encoding would call methods on the serialized object).

We should not need this anymore.